sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuOverrides.scala

```diff
@@ -1979,7 +1979,7 @@ object GpuOverrides {
         TypeSig.all,
         repeatingParamCheck = Some(RepeatingParamCheck(
           "param",
-          TypeSig.commonCudfTypes,
+          TypeSig.commonCudfTypes + TypeSig.ARRAY.nested(TypeSig.commonCudfTypes + TypeSig.STRUCT),
```

Comment:
We're adding array support, but explicitly disallowing array-of-array.  Intentional?  Also this says we support arrays of structs but not top-level structs?

Reply:
Yes, the original goal is to support only array of struct for a customer request. 

I have updated to support array of struct, array of array, top-level struct, struct of struct, struct of array, along with the integration tests, since cudf has already supported them. 

It would be nice to have them here.